app/main/views/services.py

```diff
@@ -408,10 +408,79 @@ def add_service_type_price():
     price = ServiceTypePrice()
     price.update_from_json(price_json['price'])
 
+    today = Pendulum.today()
+    tomorrow = Pendulum.tomorrow()
+    service_type_prices = ServiceTypePrice.query.filter(
+        ServiceTypePrice.supplier_code == price.supplier_code,
+        ServiceTypePrice.service_type_id == price.service_type_id,
+        ServiceTypePrice.sub_service_id == price.sub_service_id,
+        ServiceTypePrice.region_id == price.region_id
+    ).all()
+
+    # look for current price
+    current_service_type_price = None
+    current_service_type_prices = [c for c in service_type_prices if c.date_from < today and c.date_to >= today]
+    current_service_type_prices_length = len(current_service_type_prices)
+    if current_service_type_prices_length > 1:
+        return jsonify(CurrentServiceTypePrices=current_service_type_prices,
+                       msg="There is more than one current price",
+                       error=True), 400
+    elif current_service_type_prices_length == 1:
+        current_service_type_price = current_service_type_prices[0]
+
+    # look for future price
+    future_service_type_price = None
+    future_service_type_prices = [c for c in service_type_prices if c.date_from <= tomorrow and c.date_to > tomorrow]
+    future_service_type_prices_length = len(future_service_type_prices)
+    if future_service_type_prices_length > 1:
+        return jsonify(FutureServiceTypePrices=future_service_type_prices,
+                       msg="There is more than one future price",
+                       error=True), 400
+    elif future_service_type_prices_length == 1:
+        future_service_type_price = future_service_type_prices[0]
+
+    if current_service_type_price is not None and (future_service_type_price is None or
+                                                   future_service_type_price.id == current_service_type_price.id):
+        if current_service_type_price.price != price.price:
+            # Only add record when there are no future prices
+            current_service_type_price.updated_at = Pendulum.utcnow()
+            current_service_type_price.date_to = today
+
+            price.date_from = tomorrow
+            price.date_to = Pendulum(2050, 1, 1)
+
+            db.session.add(current_service_type_price)
+            db.session.add(price)
+            db.session.commit()
+            return jsonify(price=price.serializable,
+                           msg="Expired current price. Added new price",
+                           updated=True), 201
```

Comment:
same with updated - you could use a response code

Reply:
I don't quite understand this.

Reply:
its more the `update=True` or `error=True` part of the response I have a problem with

Reply:
I can't figure out why I used ```updated``` and ```error``` in the first place.
I have changed it to use status codes